test/behaviour/connection/ConnectionSteps.java

```diff
@@ -51,56 +53,44 @@
     public static Map<Session, List<CompletableFuture<Transaction>>> sessionsToTransactionsParallel = new HashMap<>();
     public static Map<CompletableFuture<Session>, List<CompletableFuture<Transaction>>> sessionsParallelToTransactionsParallel = new HashMap<>();
 
-    private static synchronized void connect_to_grakn() {
-        if (!isNull(client)) return;
-
-        System.out.println("Connecting to Grakn ...");
-
-        String address = GraknSingleton.getGraknRunner().address();
-        assertNotNull(address);
-
-        System.out.println("Establishing Connection to Grakn Core at " + address);
-        client = new GraknClient(address);
-
-        System.out.println("Connection to Grakn Core established");
-
-        assertNotNull(client);
-    }
-
     public static Transaction tx() {
         return sessionsToTransactions.get(sessions.get(0)).get(0);
     }
 
     @Given("connection has been opened")
     public void connection_has_been_opened() {
-        if (isNull(client)) {
-            connect_to_grakn();
-        }
-
         assertNotNull(client);
         assertTrue(client.isOpen());
     }
 
-    @Given("connection delete all databases")
-    public void connection_delete_all_databases() {
-        for (String database : client.databases().all()) {
-            client.databases().delete(database);
-        }
-    }
-
     @Given("connection does not have any database")
     public void connection_does_not_have_any_database() {
         assertTrue(client.databases().all().isEmpty());
     }
 
+    @Before
+    public synchronized void before() {
+        assertNull(client);
+        System.out.println("Connecting to Grakn ...");
+        String address = GraknSingleton.getGraknRunner().address();
+        assertNotNull(address);
+        System.out.println("Establishing Connection to Grakn Core at " + address);
+        client = new GraknClient(address);
+        System.out.println("Connection to Grakn Core established");
```

Comment:
Opening a new client for each scenario is fine, perhaps even desirable- but in that case let's take out the `System.out.println` statements. They'll feel like unnecessary noise when they're run in every single scenario.